### PR TITLE
fix(server): fix bug in replication on cached mode

### PR DESCRIPTION
**Pull request by adiholden**
The bug:
CVCOnBump did not serialize all the relevant buckets

The fix:
if origin bucket is stash bucket always serialize it the original bucket and its probing

### CODE REVIEW

**Comment by kostasrim** [comment]
> if origin bucket is stash bucket always serialize it the original bucket and its probing 

I would take an oath I had this fixed :face_exhaling: :facepalm:

File: src/core/dash_internal.h [hunk 1]
```diff
@@ -1544,31 +1544,9 @@ std::enable_if_t<UV, unsigned> Segment<Key, Value, Policy>::CVCOnBump(uint64_t v
...
-      SlotId slot_id = target.IterateStash(fp_hash, probe, find_stash).second;
-      if (slot_id != BucketType::kNanSlot) {
-        result_bid[result++] = target_bid;
-      }
-    }
-  };
-
-  do_fun(target, target_bid, false);
-  do_fun(probing, probing_bid, true);
+  result_bid[result++] = probing_bid;
```
Comment (kostasrim) [inline src/core/dash_internal.h:1549]: Shouldn't this assignment happen only when bucket is a `stash` bucket?

**Review by kostasrim — DISMISSED** [review]

**Comment by romange** [comment]
There is a long and detailed explanation for this function in its comments. Can you please update it @adiholden ?

File: src/core/dash_internal.h (hunk 1 above)
Comment (adiholden) [reply src/core/dash_internal.h:1549]: there is an if above checking bid < kBucketNum

**Comment by dranikpg** [comment]
SeedeV2 vs existing code - 1:0 😎

**Comment by kostasrim** [comment]
> SeederV2 vs existing code - 1:0 😎

zohan

**Comment by adiholden** [comment]
@romange can you approve?

**Review by romange — APPROVED** [review]

**Pull request merged**